File: me_pipeline/scripts/bin/Batch_wb_image_capture_volreg.m
function Batch_wb_image_capture_volreg(volume,Lpial,Lwhite,Rpial,Rwhite,outname)
%Batch_wb_image_capture_volreg(volume,Lpial,Lwhite,Rpial,Rwhite,outname)
%
%
%E.Gordon 09/18
addpath /data/nil-bluearc/GMT/Laumann/PostFreesurfer_Scripts/

origcapturefolder = '/data/nil-bluearc/GMT/Laumann/PostFreesurfer_Scripts/image_capture_template/';

[path,~,~] = fileparts(outname);
if isempty(path)
    path = pwd;
end

%rng shuffle
randnum = num2str(randi(1000000));
capturefolder = [path '/temp_image_capture_files' randnum '/'];
mkdir(capturefolder)
try copyfile([origcapturefolder '/*'],capturefolder); catch; end



%copy files
copyfile(volume,[capturefolder '/volume.nii.gz']); 
copyfile(Lpial,[capturefolder '/L.pial.surf.gii']); 
copyfile(Rpial,[capturefolder '/R.pial.surf.gii']); 
copyfile(Lwhite,[capturefolder '/L.white.surf.gii']);
copyfile(Rwhite,[capturefolder '/R.white.surf.gii']);

height = 800;
width = 2450;
system(['wb_command -volume-palette ' capturefolder '/volume.nii.gz MODE_AUTO_SCALE_PERCENTAGE -pos-percent 57 96']);
system(['wb_command -show-scene ' capturefolder '/Capture_volreg.scene 1 ' outname '.png ' num2str(width) ' ' num2str(height)]);

try rmdir(capturefolder,'s'); catch; end

File: me_pipeline/scripts/bin/batch_wb_image_capture_volreg.py

import shutil
import subprocess
from pathlib import Path
from random import randint

def batch_wb_image_capture_volreg(volume, Lpial, Lwhite, Rpial, Rwhite, outname):

    orig_capture_folder = '/data/nil-bluearc/GMT/Laumann/PostFreesurfer_Scripts/image_capture_template/'

    # Get parent of output location
    dir_name = Path(outname).parent
    if not dir_name:
        dir_name = Path.cwd()

    # Create paths for capture folder and nifti/gifti files
    rand_num = str(randint(1, 1000000))
    capture_folder_path = dir_name / f"temp_image_capture_files{rand_num}"
    volume_path = capture_folder_path / "volume.nii.gz"
    Lpial_path = capture_folder_path / "L.pial.surf.gii"
    Rpial_path = capture_folder_path / "R.pial.surf.gii"
    Lwhite_path = capture_folder_path / "L.white.surf.gii"
    Rwhite_path = capture_folder_path / "R.white.surf.gii"
    volreg_path = capture_folder_path / "Capture_volreg.scene"

    # Copy contents to capture folder
    shutil.copytree(orig_capture_folder, capture_folder_path)
    shutil.copy(volume, volume_path)
    shutil.copy(Lpial, Lpial_path)
    shutil.copy(Rpial, Rpial_path)
    shutil.copy(Lwhite, Lwhite_path)
    shutil.copy(Rwhite, Rwhite_path)

    # Run wb command
    height = "800"
    width = "2450"
    png_output_name = outname + ".png"
    subprocess.run(
        ["wb_command", "-volume-palette", volume_path, "MODE_AUTO_SCALE_PERCENTAGE", "-pos-percent", "57", "96"]
    )
    subprocess.run(["wb_command", "-show-scene", volreg_path, "1", png_output_name, height, width])

    # Recursively delete the temp capture folder
    shutil.rmtree(capture_folder_path)